### ISSUE TITLE
Refactoring and improving `AnimationTrackEditor`

### ISSUE THREAD

**Issue by TokageItLab** [godotengine/godot-proposals#3212]
### Describe the project you are working on

3D Character Game

### Describe the problem or limitation you are having in your project

We would like to implement some properties in AnimationTrack, but that is difficult to do by architecture design.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

In the current `AnimationTrackEditor`, if you want to set the properties of each track, you use the pull-down button on the right side. However, the pull-down buttons is not modularized, so it should be refactored.

<img width="582" alt="スクリーンショット 2021-08-28 0 39 04" src="https://user-images.githubusercontent.com/61938263/131153098-fb37fb58-0046-4188-9b19-a7ad00e5746e.png">

`void AnimationTrackEdit::_notification(int p_what)` in `animation_track_editor.cpp`
```cpp
// BUTTONS //

{
	Ref<Texture2D> wrap_icon[2] = {
		get_theme_icon(SNAME("InterpWrapClamp"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("InterpWrapLoop"), SNAME("EditorIcons")),
	};

	Ref<Texture2D> interp_icon[3] = {
		get_theme_icon(SNAME("InterpRaw"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("InterpLinear"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("InterpCubic"), SNAME("EditorIcons"))
	};
	Ref<Texture2D> cont_icon[4] = {
		get_theme_icon(SNAME("TrackContinuous"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("TrackDiscrete"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("TrackTrigger"), SNAME("EditorIcons")),
		get_theme_icon(SNAME("TrackCapture"), SNAME("EditorIcons"))
	};

	int ofs = get_size().width - timeline->get_buttons_width();

	Ref<Texture2D> down_icon = get_theme_icon(SNAME("select_arrow"), SNAME("Tree"));

	draw_line(Point2(ofs, 0), Point2(ofs, get_size().height), linecolor, Math::round(EDSCALE));

	ofs += hsep;
	{
		//callmode

		Animation::UpdateMode update_mode;

		if (animation->track_get_type(track) == Animation::TYPE_VALUE) {
			update_mode = animation->value_track_get_update_mode(track);
```

Also, visually, only a maximum of three buttons can be placed, but there are not always less than three properties that a track has. For example, there are https://github.com/godotengine/godot/pull/49099 and https://github.com/godotengine/godot-proposals/issues/2831 as use cases, and these will probably need to be added to the track properties, but there may be other options that should be added.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In conclusion, we should make tracks selectable in `AnimationTrackEditor` and show the information of the selected track in the inspector.

![2021-08-28_0 47 18](https://user-images.githubusercontent.com/61938263/131155715-feb492e4-1013-4711-bac9-fe64ced645c3.png)

Also, it might be better show the "track information" and "key information" with separating the inspector by label when you select keyframe.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, `AnimationTrackEditor` is in the core.

### Is there a reason why this should be core and not an add-on in the asset library?

This is needed for future implementations of AnimationTrack properties.

**Comment by NathanLovato** [comment]
I totally agree. Actually, the bezier editor is supposed to get the ability to select one (or more) tracks. It makes sense also to have the ability to select multiple tracks at once and change their attributes, like muting several, but perhaps also moving forward hiding and unhiding them from the editor as the dope sheet view easily gets cluttered.

**Comment by LightningAA** [comment]
Should this maybe be a comment on https://github.com/godotengine/godot-proposals/issues/3141 instead of a separate issue? Or is this different?

**Comment by NathanLovato** [comment]
This is different as the other proposal doesn't deal with the properties of an animation track, only improving the interface of the bezier editor.

**Comment by nonunknown** [comment]
AS was discussed in the meeting one improvement is the ability to copy/paste to clipboard a track value to paste between animations of the same animationplayer:

`stance`
-------------------
[Sprite:Position] * * *
[Collisionbox2D:Shape] *    *    *  <- copy here
-------------------

`walk`
-------------------
[Sprite:Position] * * *`
-------------------

Pressing CTRL-V in the walk animation will past the [Collisionbox2D:Shape] values on it!

**Comment by NathanLovato** [comment]
That should probably be another proposal as it's not related to the anim track editor - that should probably work for bezier curves too for instance.

And that's probably not trivial either. Like at times, you'll want to copy just the values of keyframes into other tracks (even on different nodes, for example, copy an animation that rotates one arm of a character onto another arm) and at other times insert the corresponding animation tracks, that is, copy the whole keyframes.

**Comment by TokageItLab** [comment]
@nonunknown There is a difference with the problem you mentioned. What I mean is that the AnimationTree implicitly changes the volume of the AudioStreamPlayer.

What @lyuma and I were discussing before was that we should have a track property to indicate whether or not to change the volume implicitly, and that we should be able to create a separate Bezier track to manually change the volume.

However, it's certainly common in DAWs to display the audio waveform in the background or on top of the volume bezier curve, so whether it's necessary to do that in Godot is something that needs to be discussed, but I think that it's worth making the proposal.

**Comment by nonunknown** [comment]
ohh right! got it, I'll make a proposal for that then!
